### PR TITLE
feat: screen for empty observation and observation list

### PR DESCRIPTION
**Pull request by cimigree**
Related to #50 
This addresses but does not complete #50.
This PR adds the Main screen that a user will go to after they join a project. It includes the Empty Observation State and the Observation List View.

**Some decisions I made.**
- This PR does not include any logic related to putting the observations on the map. It is already quite big!
- This PR does not create test tracks. It only creates test Observations. 
- After consulting with Andrew, I decided to create those observations similarly to how it is done in mobile. It happens here when someone creates a project. This will just be for us to use during development and can be removed as soon as desktops become discoverable and someone can join a project.
- I am directly using the hooks from comapeo/core-react when they are needed instead of creating separate files to hold the (very simple and straightforward) queries. Hope that makes sense.
- I copied over the logic from mobile to match presets.
- For images, if there is an svg in mobile, I just copied it over. However, if it was easy to get the png from FIgma, I copied the png from there. Sorry there is a mix of svgs and pngs. In one case, though, with the add_person, I used an svg so it could have different colors.
- I decided to name the main route "main" instead of "tab1." Several changes and changed files in here are related to that name change. It felt more semantic and meaningful to me. Hope that is ok.
- I expanded the width of the column on the left that is going to hold everything (to fit the buttons' text, but it made sense to me for it to be a bit bigger since it is where all the info will be).
- I tried to keep the logic in the map.main screen and keep the component screens more presentational. The small exception is that I did not want to pass down the projectId because it had to go through the list view without being used, so I just fetch it in the list item where it is needed for the preset. In general, I tried not to pass down too many props, but I left the navigation in the main screen which is a lot of the props being passed down. There is a tension between keeping the logic out of presentational components and passing too many props.
- I added the DARK_TEXT (#333333) as the main text color. In looking at Figma, that seemed to be the color for all of the text. 
- I am trying to add a test to check which version the main screen renders (empty or list) but it is a nightmare, so am putting that off for a separate PR.

I am getting all kinds of errors related to translating the presets and translation, even though I am doing it the same way as mobile. I am not sure why or how to fix that.

Observation List
---
Screenshot 2025-01-15 at 4 49 05 PM

Empty Screen
---
Screenshot 2025-01-15 at 4 51 40 PM

### CODE REVIEW

**Review by achou11 — CHANGES_REQUESTED** [review]
Initial review but overall looking good so far!

> After consulting with Andrew, I decided to create those observations similarly to how it is done in mobile. It happens here when someone creates a project. This will just be for us to use during development and can be removed as soon as desktops become discoverable and someone can join a project.

I made a note of this in a comment, but I would suggest making the data generation opt-in so that we can still work with an empty state.

> I am directly using the hooks from comapeo/core-react when they are needed instead of creating separate files to hold the (very simple and straightforward) queries. Hope that makes sense.

This is preferable! I don't imagine moving away from our own thing library so the pattern you're alluding to is less relevant.

> For images, if there is an svg in mobile, I just copied it over. However, if it was easy to get the png from FIgma, I copied the png from there. Sorry there is a mix of svgs and pngs. In one case, though, with the add_person, I used an svg so it could have different colors.

yeah i think we'd ideally have SVGs in most cases? Wouldn't worry too much for now unless it looked really off for whatever reason. Would be good to create an issue to track which ones might need replacing though.

> I decided to name the main route "main" instead of "tab1." Several changes and changed files in here are related to that name change. It felt more semantic and meaningful to me. Hope that is ok.

I don't mind the renaming in general, although not sure what the actual name should be. generally agree that these tab names should be more meaningful, so i guess "main" is a maybe a slight improvement over "tab1"? Doesn't help that these designs could change and thus make any name that's chosen potentially confusing.

> I expanded the width of the column on the left that is going to hold everything (to fit the buttons' text, but it made sense to me for it to be a bit bigger since it is where all the info will be).

Yeah we need to choose a minimum window size that we want to commit to, which I think will inform the minimum+maximum size of that panel.

> I tried to keep the logic in the map.main screen and keep the component screens more presentational. The small exception is that I did not want to pass down the projectId because it had to go through the list view without being used, so I just fetch it in the list item where it is needed for the preset.

yeah this is more related to the fact that the active project ID should really be part of the URL so that these kinds of issues are no longer relevant.

> I am trying to add a test to check which version the main screen renders (empty or list) but it is a nightmare, so am putting that off for a separate PR.

Yeah i think the test setup needs some attention to make things less tedious and more consistent. Some of that work was being done in a separate PR but might be worth me taking a look at it.

File: src/renderer/src/components/Observations/EmptyState.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+			<DividerLine />
+			<LowerContainer>
+				<Circle>
+					<StyledImage src={EmptyStateImage} alt="Empty Observations List" />
+				</Circle>
+				<Text
+					kind="body"
+					style={{
+						marginTop: 18,
+					}}
```
[the remark this answers is not in the document]
Comment (cimigree) [reply src/renderer/src/components/Observations/EmptyState.tsx:90]: Ok I removed as many as I could to outer containers with gap and/ or padding

File: src/renderer/src/components/Observations/ObservationListItem.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+		: new Date()
+
+	const photoAttachment = observation.attachments.find(
+		(att) => att.type === 'photo',
+	)
+
+	return (
+		<Container onClick={onClick}>
+			<TextContainer>
+				<Text style={{ fontWeight: 500 }}>
```
[the remark this answers is not in the document]
Comment (cimigree) [reply src/renderer/src/components/Observations/ObservationListItem.tsx:57]: bold is 700, which is why I did this. The designs call for 500. Should I ignore the designs?

File: src/renderer/src/components/Observations/ObservationListView.tsx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+	return (
+		<Container>
+			<ContentWrapper>
+				<ButtonsRow>
+					<Button
+						variant="contained"
+						style={{
+							backgroundColor: DARK_ORANGE,
+							color: WHITE,
+							minWidth: 170,
```
[the remark this answers is not in the document]
Comment (cimigree) [reply src/renderer/src/components/Observations/ObservationListView.tsx:78]: I just used flex: 1 instead so the margins on the side would be the same. Hope that makes sense.

File: src/renderer/src/components/Observations/ObservationListView.tsx [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+	return (
+		<Container>
+			<ContentWrapper>
+				<ButtonsRow>
+					<Button
+						variant="contained"
+						style={{
+							backgroundColor: DARK_ORANGE,
+							color: WHITE,
```
[the remark this answers is not in the document]
Comment (cimigree) [reply src/renderer/src/components/Observations/ObservationListView.tsx:77]: I took your suggestion about avoiding inline styling, but rather than overriding secondary (since secondary is a different orange color than what is in the designs) or adding a new palette color, I opted to define a custom MuiButton variant called "darkOrange". This keeps our existing color definitions untouched and lets us reuse the style simply with variant="darkOrange". I am thinking of these variants as being something for now which we can update when we get more guidance from design and different button types.

File: src/renderer/src/routes/Onboarding/CreateProjectScreen.tsx [hunk 5]
```diff
@@ -167,7 +170,18 @@ function CreateProjectScreenComponent() {
...
+						{
+							onSuccess: () => {
+								navigate({ to: '/main' })
+							},
+							onError: (err) => {
+								console.error('Error creating test data', err)
+								navigate({ to: '/main' })
+							},
+						},
+					)
```
[the remark this answers is not in the document]
Comment (cimigree) [reply src/renderer/src/routes/Onboarding/CreateProjectScreen.tsx:184]: Ok I put a button in the settings tab and then navigate back to the main tab after they are created. Quick and dirty fix for now. Hope that is ok.

File: src/renderer/src/Theme.ts [hunk 6]
```diff
@@ -33,17 +44,24 @@ const theme = createTheme({
...
 			fontSize: '1.5rem',
 			fontWeight: 600,
+			...commonTextStyle,
 		},
 	},
 	palette: {
+		text: {
+			primary: DARK_TEXT,
+			secondary: DARK_GREY,
+		},
```
[the remark this answers is not in the document]
Comment (cimigree) [reply src/renderer/src/Theme.ts:64]: I took it out but I end up with weird colors in here:
So not sure how to make that right.
Screenshot 2025-01-20 at 4 26 40 PM

[the remark this answers is not in the document]
Comment (achou11) [reply src/renderer/src/Theme.ts]: hm yeah, looks like it's related to the text here: 

https://github.com/digidem/comapeo-desktop/blob/feat/main-screen/src/renderer/src/components/Onboarding/OnboardingTopMenu.tsx#L96

The `Step` component that wraps it defines a `color` which _I think_ used to cascade to the `Text` component within it since it wasn't defining its own color prop prior to my suggestion?

https://github.com/digidem/comapeo-desktop/blob/4b5966544e9717e42a86b45f2dad447b5b5058b6/src/renderer/src/components/Onboarding/OnboardingTopMenu.tsx#L38

Somewhat imperfect, but for the sake of not needing to make too many changes right now, I would suggest the following:

- remove any text-related styling that's defined by the the `Step`, notably `color ` and `fontWeight`.
- update the `Text` component in that `OnboardingTopMenu` component to inline the white color styling i.e. `style={{ color: currentStep !== step ? WHITE : undefined }}`

As an eventual follow-up, we should probably define a text definition in the theme that basically equates to "inverted" coloring i.e. light text on darker backgrounds (since primary and secondary currently work for lighter backgrounds)

**Review by achou11 — APPROVED** [review]
Should've made the comment as part of the review but aside from that, looks good to me!

- [ ] https://github.com/digidem/comapeo-desktop/pull/84#discussion_r1924076021
